src/Service.Host/client/src/components/StockPools.js

```diff
@@ -0,0 +1,306 @@
+
+    const {
+        send: updateStockPool,
+        isLoading: updateLoading,
+        errorMessage: updateError,
+        putResult: updateResult
+    } = usePut(itemTypes.stockPools.url, true);
+
+    const {
+        send: createStockPool,
+        createStockPoolLoading,
+        errorMessage: createStockPoolError,
+        postResult: createStockPoolResult,
+        clearPostResult: clearCreateStockPool
+    } = usePost(itemTypes.stockPools.url);
+
+    useEffect(() => {
+        setSnackbarVisible(!!updateResult || !!createStockPoolResult);
+    }, [createStockPoolResult, updateResult]);
+
+    useEffect(() => {
+        setStockPools(stockPoolResult);
+    }, [stockPoolResult]);
+
+    const [searchDialogOpen, setSearchDialogOpen] = useState({
+        forRow: null,
+        forColumn: null
+    });
+    const searchRenderCell = params => (
+        <>
+            <GridSearchIcon
+                style={{ cursor: 'pointer' }}
+                onClick={() =>
+                    setSearchDialogOpen({
+                        forRow: params.id,
+                        forColumn: params.field
+                    })
+                }
+            />
+            {params.value}
+        </>
+    );
+
+    const addNewRow = () => {
+        setStockPools([
+            ...stockPools,
+            {
+                accountingCompany: null,
+                availableToMrp: 'Y',
+                bridgeId: null,
+                dateInvalid: null,
+                defaultLocation: null,
+                links: null,
+                sequence: null,
+                stockCategory: null,
+                stockPoolCode: '',
+                stockPoolDescription: null,
+                storageLocation: null,
+                creating: true,
+                id: stockPools.length + 1
+            }
+        ]);
+        setCreating(true);
+    };
+
+    const processRowUpdate = newRow => {
+        const updatedRow = { ...newRow, updated: true };
+        console.log(updatedRow);
+        setStockPools(prevRows =>
+            prevRows.map(row =>
+                row.stockPoolCode === newRow.stockPoolCode || row.stockPoolCode === ''
+                    ? updatedRow
+                    : row
+            )
+        );
+        return newRow;
+    };
+
+    const renderSearchDialog = c => {
+        const handleClose = () => {
+            setSearchDialogOpen({ forRow: null, forColumn: null });
+        };
+
+        const handleSearchResultSelect = selected => {
+            const currentRow = stockPools?.find(r => r.id === searchDialogOpen.forRow);
+            let newRow = {
+                ...currentRow,
+                updated: true,
+                defaultLocation: selected.locationId,
+                defaultLocationName: selected.locationCode
+            };
+            c.searchUpdateFieldNames?.forEach(f => {
+                newRow = { ...newRow, [f.fieldName]: selected[f.searchResultFieldName] };
+            });
```

Comment:
same here - overly complicating things -  to keep things clearer/easier to follow just directly update the newRow fields you want in here as opposed to having to fish some specifications out of the columns defintion